Update plugin to add support for server-to-server oauth app authentication (#35)
turbot/steampipe-plugin-zoom: docs/index.md

### Credentials

| Item        | Description                                                                                                                                                                                                                                                  |
| ----------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |

> | Credentials | [Create a Server-to-Server OAuth app](https://developers.zoom.us/docs/internal-apps/create/) to get the Account ID, Client ID and Client Secret or [create an SDK App](https://marketplace.zoom.us/docs/guides/build/sdk-app) to get the API key and secret. |

Reviewer: ```suggestion
| Credentials | [Create a Server-to-Server OAuth app](https://developers.zoom.us/docs/internal-apps/create/) to get the Account ID, Client ID and Client Secret. |
```

Author: updated

  ## Server-to-Server OAuth app credentials

  # `account_id`(required) - The Zoom account ID.
  # Can also be set with the ZOOM_ACCOUNT_ID environment variable.
  # account_id    = "Xt1aUD4WQ56w7hDhVbtDp"

  # `client_id`(required) - The Zoom Client ID provided by Server-to-Server OAuth app.
  # Can also be set with the ZOOM_CLIENT_ID environment variable.
  # client_id    = "MZw2piRfTsOdpwx2Dh5U"

  # `client_secret`(required) - The Zoom Client Secret provided by Server-to-Server OAuth app.
  # Can also be set with the ZOOM_CLIENT_SECRET environment variable.
  # client_secret    = "04tKwHgFGvwB1M4HPHOBFP0aLHYqUE"

  ## SDK/JWT app credentials

  # `api_key`(required)   = The Zoom API key provided by SDK/JWT OAuth app.
  # Can also be set with the ZOOM_API_KEY environment variable.
  # api_key    = "LFMU3oagTjO8_5sYKQVe"

  # `api_secret`(required) = The Zoom API secret provided by SDK/JWT OAuth app.
  # Can also be set with the ZOOM_API_SECRET environment variable.
  # api_secret = "PKS96L69nWSFK2y0A07R2k7xGryVbcWiem"
}
```

## Configuring Zoom Credentials

> You may specify the AccountID, ClientID and ClientSecret to authenticate:

Reviewer: ```suggestion
### Server-to-Server OAuth Application

You may specify the account ID, client ID, and client secret to authenticate:
```

Author: updated

- `account_id`: The Zoom account ID.
- `client_id`: The Zoom Client ID provided by Server-to-Server OAuth app.
- `client_secret`: The Zoom Client Secret provided by Server-to-Server OAuth app.

```hcl
connection "zoom" {
  plugin        = "zoom"
  account_id    = "Xt1aUD4WQ56w7hDhVbtDp"
  client_id     = "MZw2piRfTsOdpwx2Dh5U"
  client_secret = "04tKwHgFGvwB1M4HPHOBFP0aLHYqUE"
}
```

> or you may specify the APIKey and APISecret to authenticate:

Reviewer: ```suggestion
### JWT Application

Note: JWT applications are deprecated as of June 1, 2023 and will be entirely disabled on September 1, 2023. We recommend migrating to Server-to-Server OAuth applications.

You may specify the API key and API secret:
```

Author: updated

### Credentials

| Item        | Description                                                                                                                                                                                                           |
| ----------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| Credentials | [Create a Server-to-Server OAuth app](https://developers.zoom.us/docs/internal-apps/create/) to get the Account ID, Client ID and Client Secret.                                                                      |

> | Permissions | Server-to-Server OAuth apps or SDK / JWT apps can access all Zoom APIs.                                                                                                                                               |

Reviewer: Can scopes be set on OAuth or JWT apps now?

Author: Yes, we can set the scope on OAuth or JWT apps. I have updated the message; please let me know if it looks good or if there are any changes.

| Radius      | Each connection represents a single Zoom account.                                                                                                                                                                     |

> | Resolution  | 1. Credentials explicitly set in a steampipe config file (`~/.steampipe/config/zoom.spc`)<br />2. Credentials specified in environment variables, e.g., `ZOOM_ACCOUNT_ID`, `ZOOM_CLIENT_ID` and `ZOOM_CLIENT_SECRET`. |

Reviewer: ```suggestion
| Resolution  | 1. Credentials explicitly set in a Steampipe config file (`~/.steampipe/config/zoom.spc`)<br />2. Credentials specified in environment variables, e.g., `ZOOM_ACCOUNT_ID`, `ZOOM_CLIENT_ID` and `ZOOM_CLIENT_SECRET`. |
```

Author: updated

### Configuration

Installing the latest zoom plugin will create a config file (`~/.steampipe/config/zoom.spc`) with a single connection named `zoom`:

Configure your account details in `~/.steampipe/config/zoom.spc`:

```hcl
connection "zoom" {
  plugin = "zoom"

  # Zoom API credentials are available to users with Developer role in the account.
  # You need to create a Server-to-Server OAuth app(https://developers.zoom.us/docs/internal-apps/create) or a SDK/JWT APP (https://marketplace.zoom.us/docs/guides/build/sdk-app)to get the credentials.
  # It is recommended that you create Server-to-Server OAuth as JWT app is deprecated On June 1, 2023 and will be disabled on September 1, 2023. https://developers.zoom.us/docs/internal-apps/jwt-faq/

>   ## Server-to-Server OAuth app credentials

Reviewer: ```suggestion
  # Server-to-Server OAuth app credentials
```

Author: updated

  # `account_id`(required) - The Zoom account ID.
  # Can also be set with the ZOOM_ACCOUNT_ID environment variable.
  # account_id = "Xt1aUD4WQ56w7hDhVbtDp"

  # `client_id`(required) - The Zoom Client ID provided by Server-to-Server OAuth app.
  # Can also be set with the ZOOM_CLIENT_ID environment variable.
  # client_id = "MZw2piRfTsOdpwx2Dh5U"

  # `client_secret`(required) - The Zoom Client Secret provided by Server-to-Server OAuth app.
  # Can also be set with the ZOOM_CLIENT_SECRET environment variable.
  # client_secret = "04tKwHgFGvwB1M4HPHOBFP0aLHYqUE"

>   ## SDK/JWT app credentials

Reviewer: ```suggestion
  # SDK/JWT app credentials
```

Author: updated

}
```

or through environment variables

> The Zoom plugin will use the Zoom environment variable to obtain credentials **only if the `account_id`,`client_id`, and`client_secret` or `api_key` and `api_secret` is not specified** in the connection:

Reviewer: ```suggestion
The Zoom plugin will use the Zoom environment variable to obtain credentials **only if the `account_id`, `client_id`, and `client_secret` or `api_key` and `api_secret` arguments are not specified** in the connection:
```

Please watch out for small syntax/grammar errors, you can use tools like Grammarly or ChatGPT to help with proofreading.

Author: updated.
